lib/CMakeLists.txt

```diff
@@ -30,6 +30,11 @@
 
 set(CMAKE_C_FLAGS "-Wall -Wextra -Werror -fPIC")
 set(CMAKE_C_FLAGS_DEBUG "-O0 -ggdb")
+#
+# NB: this intentionally does not define NDEBUG: we want our assert()s even in a
+# release build.
+#
+set(CMAKE_C_FLAGS_RELEASE "-O3")
```

Comment:
I think NDEBUG should be undefined from top level CMakeLists.txt. So that this rule is applicable all over the repo.

Reply:
Thanks, yes!